main.py

```diff
@@ -481,6 +484,92 @@ def delete_interface_metadata(self, request: Request) -> JSONResponse:
+        content = get_json_or_400(request, self.controller.loop)
+        ranges: list[Union[int, list[int]]] = content["tag_ranges"]
+        tag_type = self._get_tag_type(content)
+        ranges[0] = self.check_non_range(ranges[0])
+        for i in range(1, len(ranges)):
+            ranges[i] = self.check_non_range(ranges[i])
+            if ranges[i][0] > ranges[i][1]:
+                detail = f"The range {ranges[i]} is not ordered"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] > ranges[i][0]:
+                detail = f"tag_ranges is not ordered. {ranges[i-1][1]}"\
+                         f" is higher than {ranges[i][0]}"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] == ranges[i][0] - 1:
+                detail = f"tag_ranges has an unnecessary partition. "\
+                         f"{ranges[i-1][1]} is before to {ranges[i][0]}"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] == ranges[i][0]:
+                detail = f"tag_ranges has repetition. {ranges[i-1]}"\
+                         f" have same values as {ranges[i]}"
+                raise HTTPException(400, detail=detail)
+        if ranges[-1][1] > 4095:
+            detail = "Maximum value for tag_ranges is 4095"
+            raise HTTPException(400, detail=detail)
+        if ranges[0][0] < 1:
+            detail = "Minimum value for tag_ranges is 1"
+            raise HTTPException(400, detail=detail)
+        return ranges, tag_type
+
+    @rest('v3/interfaces/{interface_id}/tag_ranges', methods=['POST'])
+    @validate_openapi(spec)
+    def add_tag_range(self, request: Request) -> JSONResponse:
+        """Add/modify tag range"""
+        interface_id = request.path_params["interface_id"]
+        ranges, tag_type = self._get_tag_ranges(request)
+        interface = self.controller.get_interface_by_id(interface_id)
+        if not interface:
+            raise HTTPException(404, detail="Interface not found")
+        try:
+            interface.set_tag_ranges(ranges, tag_type)
+            interface.notify_link_available_tags(self.controller)
+        except KytosResizingAvailableTagError as err:
+            detail = f"The new tag_ranges cannot be applied {err}"
+            raise HTTPException(400, detail=detail)
+
+        raise HTTPException(200, detail="Operation Successful")
+
+    @rest('v3/interfaces/{interface_id}/tag_ranges', methods=['DELETE'])
+    @validate_openapi(spec)
+    def delete_tag_range(self, request: Request) -> JSONResponse:
+        """Delete tag range"""
+        interface_id = request.path_params["interface_id"]
+        params = request.query_params
+        tag_type = self._get_tag_type(params)
+        interface = self.controller.get_interface_by_id(interface_id)
+        if not interface:
+            raise HTTPException(404, detail="Interface not found")
+        interface.remove_tag_ranges(tag_type)
```

Comment:
Looking again at this deletion, it's more behaving like setting a default tag_ranges than an actual deletion. I mean in the issue discussion it was also discussed regarding deleting a pair of tag ranges, but based on how POST has been structured, POST can be reused to reset whatever is needed including deleting an existing tag range interval if you send again all the other intervals together. With that said, it's reasonable, but let's make it crystal clear what the deletion does, which is resetting the entire tag_ranges for a given tag_type. So if you could update docstrings and API descriptions that'd be great.

Reply:
Updated docstrings